backend/src/Edna/Web/API.hs

```diff
@@ -32,3 +44,157 @@ type EdnaAPI =
 
 ednaAPI :: Proxy EdnaAPI
 ednaAPI = Proxy
+
+-- | Endpoints necessary to implement file uploading.
+data FileUploadEndpoints route = FileUploadEndpoints
+  { -- | Parse the file and return its summary for preview.
+    fueParseFile :: route
+      :- "parse"
+      :> Summary "Parse the file and return its summary for preview"
+      :> MultipartForm Mem (MultipartData Mem)
+      :> Get '[JSON] FileSummary
+
+  , -- | Upload the file with some methodology and project.
+    fueUploadFile :: route
+      :- "upload"
+      :> Summary "Upload the file with some methodology and project"
+      :> ReqBody '[JSON] FileUploadReq
+      :> MultipartForm Mem (MultipartData Mem)
+      :> Post '[JSON] FileSummary
+  } deriving stock (Generic)
+
+type FileUploadAPI = ToServant FileUploadEndpoints AsApi
+
+-- TODO: pagination and sorting are just stubs for now (everywhere).
+-- Most likely we will use @servant-util@ to implement them,
+-- but let's do it later.
+
+-- | Endpoints related to projects.
+data ProjectEndpoints route = ProjectEndpoints
+  { -- | Add a new project.
+    peAddProject :: route
+      :- "project"
```

Comment:
Move it to `EdnaEndpoints` as you made for `"file"` and also move `"methodology"`, `"compound"` and `"targets"`

Reply:
It's not so easy because we have `project` and `projects` here, there 2 different getters. The same is true in other structures. So I can't move it easily. I can keep only `project` endpoints here and extract `projects`, but then we will have more fields in `EdnaEndpoints` which is not very good. So IMO it won't be an improvement, it will have one advantage and one disadvantage.